### ISSUE TITLE
is there anyway to change the age u have to be to get this mod to work?

### ISSUE THREAD

**Issue by plankys** [NickWildish/Mas-NSFW-Submod#157]
i put myself very young as a joke, i know you cant change ur birthday in the game but is there a way to change the age u have to be in game for this to work?

**Comment by Someone2124** [comment]
i dont think so mann, unless youre cool running the risk of her Leaving you then probably edit the main persistent file (the one thats literally named persistent) and editing your age from there?

idk im not risking that all though im not really 24 yet so maayyybeeeee i could consider editing mine.

**Comment by WinnerWind** [comment]
Hi there!
This post is old, I know, but I wrote a guide [on my website](https://winnerwind.github.io/blog-dump/tutorials/mas-birthday.html) to do this. I **do not reccomend** doing this as doing so may have unintended side effects, and *may* require time travel.

**Comment by Justformas** [comment]
This method to change birthday in-game worked for me when I tried it, but it only works once per label. You may want to back up your persistent files just in case. Awhile ago I found this here in the official MAS code: https://github.com/Monika-After-Story/MonikaModDev/blob/master/Monika%20After%20Story/game/script-story-events.rpy#L1879-L1889

Note: The first method might not work for you, but the second method should. It might be better to just use the second method. Both methods involve creating a new text file and titling it birthday.rpy or whatever, so that it's an .rpy file. 

For the first method, copy and paste this text with all the spaces into the .rpy file. You could also copy it from the link instead but would have to remove the hashtags at the start of each line and add `_select `after `mas_bday_player_bday`:

```
init 5 python:
    addEvent(
        Event(
            persistent.event_database,
            eventlabel="mas_bday_player_bday_select",
            conditional=(
                "renpy.seen_label('monika_birthday')"
            ),
            action=EV_ACT_QUEUE
        )
    )
```

The second method is nearly the same as the first. Just that you need a new label and code that effectively brings you back to the label `mas_bday_player_bday_select`. You can use the disabled label there in the game's code for this. For example you could use this code (with all the spaces) for your .rpy file:


```
init 5 python:
    addEvent(
        Event(
            persistent.event_database,
            eventlabel="mas_bday_player_bday",
            conditional=(
                "renpy.seen_label('monika_birthday')"
            ),
            action=EV_ACT_QUEUE
        )
    )

label mas_bday_player_bday:
        jump mas_bday_player_bday_select
```

Then (for either method) save this file and either put it in the MAS "game" folder or the submods folder (game folder might not work with version 0.13.0 and later). Start the game and after some moments, Monika should bring up the topic where she asks when your birthday is, allowing you to change it. Then say goodbye to Monika and after the game closes, remove this .rpy file and the .rpyc version of it that was created from your MAS directory.

For the second method only, you could also change `eventlabel="mas_bday_player_bday",` and `label mas_bday_player_bday:` to whatever new label name you want as long as they match ("new" meaning it isn't already used in the game's code and isn't one that you already used), which allows you to change birthday additional times if needed. I recommend using a very unique label name that is unlikely to get added/used in a future MAS update, so you don't run into potential issues. There are more efficient methods to keep changing birthday, but hopefully you shouldn't have to.
